feat: Pass extra request handler context to tool execution

AyushWalekar opened the pull request:
## Description

Updated the server to pass the 'extra' context from the request handler to the tool's execute method. Modified types to include 'extra' in the execute options, enabling tools to access additional request context if needed.

Here is MCP contract for ref:
https://github.com/modelcontextprotocol/typescript-sdk/blob/71d309d0d8e3a2f45e3fce5b30178531325073df/src/shared/protocol.ts#L643-L665

```ts
    /**
     * Registers a handler to invoke when this protocol object receives a request with the given method.
     *
     * Note that this will replace any previous request handler for the same method.
     */
    setRequestHandler<T extends ZodObject<{
        method: ZodLiteral<string>;
    }>>(requestSchema: T, handler: (request: z.infer<T>, extra: RequestHandlerExtra<SendRequestT, SendNotificationT>) => SendResultT | Promise<SendResultT>): void;
    /**
```

## Related Issue(s)

https://github.com/mastra-ai/mastra/issues/5510

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [x] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works 
  - if the change is acceptable, will add tests, if required.

AyushWalekar committed:
feat: Pass extra request handler context to tool execution

Updated the server to pass the 'extra' context from the request handler to the tool's execute method. Modified types to include 'extra' in the execute options, enabling tools to access additional request context if needed.

NikAiyer reviewed (dismissed):
Thanks for this PR! Can you also add some tests for this functionality?

TylerBarnes committed:
Merge branch 'main' into fix-mcp-tool-call-pass-auth-props

AyushWalekar committed:
chore: added mcp tool extras test-case

AyushWalekar committed:
refactor: streamline auth information handling in MCPServer tests

- Removed redundant checks for tool existence in the auth test case.
- Updated the test to utilize a mock request structure for better clarity.
- Enhanced assertions to validate the response structure and content more thoroughly.

wardpeet reviewed (approved):
Thank you!

AyushWalekar committed:
Merge branch 'main' into fix-mcp-tool-call-pass-auth-props

wardpeet committed:
add changeset
